core/src/main/java/io/opentelemetry/android/OpenTelemetryRum.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android
+
+import android.app.Application
+import io.opentelemetry.android.config.OtelRumConfig
+import io.opentelemetry.android.session.SessionProvider
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.sdk.OpenTelemetrySdk
+
+/**
+ * Entrypoint for the OpenTelemetry Real User Monitoring library for Android.
+ */
+interface OpenTelemetryRum {
+    /**
+     * Get a handle to the instance of the OpenTelemetry API that this
+     * instance is using for instrumentation.
+     */
+    val openTelemetry: OpenTelemetry
+
+    /**
+     * Get the client session ID associated with this instance of the RUM instrumentation library.
+     * Note: this value will change throughout the lifetime of an application instance, so it is
+     * recommended that you do not cache this value, but always retrieve it from here when needed.
+     */
+    fun getRumSessionId(): String
+
+    /**
+     * Emits an event with the specified name and body.
+     *
+     * This method serves as a convenience overload that emits an event with an empty set of
+     * attributes.
+     *
+     * @param eventName The name of the event to emit.
+     * @param body The body of the event, typically containing additional data.
+     */
+    fun emitEvent(
+        eventName: String,
+        body: String,
+    ) {
+        emitEvent(eventName, body, Attributes.empty())
+    }
+
+    /**
+     * Emits an event with the specified name and attributes.
+     *
+     * This method serves as a convenience overload that emits an event with an empty body.
+     *
+     * @param eventName The name of the event to emit.
+     * @param attributes The attributes associated with the event.
+     */
+    fun emitEvent(
+        eventName: String,
+        attributes: Attributes = Attributes.empty(),
+    ) {
+        emitEvent(eventName, "", attributes)
+    }
+
+    /**
+     * Emits an event with the specified name, body, and attributes.
+     *
+     * Implementations of this method should define how the event is handled and recorded.
+     *
+     * @param eventName The name of the event to emit.
+     * @param body The body of the event, typically containing additional data.
+     * @param attributes The attributes associated with the event, providing metadata.
+     */
+    fun emitEvent(
+        eventName: String,
+        body: String,
+        attributes: Attributes,
+    )
```

Comment:
I think all 3 of these `emitEvent` methods can be collapsed into a single method with defaults... `body = ""` and `attributes = Attributes.empty()`...probably with `@JvmOverloads` or whatever for backwards compatibility?

Reply:
I've collapsed this into a single method with defaults. Unfortunately `@JvmOverloads` isn't allowed on functions within interfaces, so folks would need to use named parameters if they wanted to omit `body` but supply `attributes`.